src/OrbitGl/ScopeTreeTest.cpp

```diff
@@ -28,6 +29,40 @@ TestScope* CreateScope(uint64_t start, uint64_t end) {
   return scope_buffer.back().get();
 }
 
+uint64_t GetFakeTimeStamp() {
+  static uint64_t count = 0;
+  return ++count;
+}
+
+struct ScopeTimer {
+  ScopeTimer(std::vector<TestScope*>* scopes, size_t max_nodes) {
+    start = GetFakeTimeStamp();
+    max_num_nodes = max_nodes;
+    scope_buffer = scopes;
+  }
+  ~ScopeTimer() {
+    if (scope_buffer->size() < max_num_nodes) {
+      scope_buffer->push_back(CreateScope(start, GetFakeTimeStamp()));
+    }
+  }
+  uint64_t start;
+  size_t max_num_nodes;
+  std::vector<TestScope*>* scope_buffer;
+};
+
```

Comment:
A little comment could help reader to understand, something like:
"Create a test tree using DFS-algorithm to assure that for each pair of intervals, one is completely inside of the other (father-children) or they don't overlap at all"

Reply:
Done, thanks.